contracts/src/interfaces/IHyperdriveRegistry.sol

```diff
@@ -2,10 +2,58 @@
 pragma solidity 0.8.20;
 
 interface IHyperdriveRegistry {
-    /// @notice Allows anyone to get the info for a hyperdrive instance.
-    /// @param _hyperdriveInstance The hyperdrive instance address.
-    /// @return The uint256 value set by governance.
+    /// @dev The info related to each Hyperdrive instance.
+    struct HyperdriveInfo {
+        /// @dev Data about the instance. Different registries can utilize
+        ///      different schemas for these values.
+        uint256 data;
+        /// @dev The factory that deployed this instance.
+        address factory;
+        /// @dev The base token of the instance.
+        IERC20 baseToken;
+        /// @dev The vault shares token of the instance.
+        IERC20 vaultSharesToken;
+        /// @dev The name of the instance.
+        string name;
+        /// @dev The version of the instance.
+        string version;
```

Comment:
Why are we including this when it can be obtained from the instance itself?

Don't get me wrong, I'm all for wrapper functions or things that make obtaining instance information more convenient... I'm just not sure why it's beneficial to store these.

Reply:
This is the struct that is returned in the getter. It's not actually stored. The struct used for storage is `HyperdriveInfoInternal`.

Reply:
Very excited for the name field in the registry getter, my current method of generating a name is 3 contract calls per pool, so combining that to one call is great